stubs/docutils/docutils/io.pyi

```diff
@@ -0,0 +1,67 @@
+from typing import Any, ClassVar, Optional, Union
+from typing_extensions import Literal
+
+from docutils import TransformSpec
+
+__docformat__: str
+
+class InputError(IOError): ...
+class OutputError(IOError): ...
+
+def check_encoding(stream: Any, encoding: str) -> Optional[bool]: ...
+
+class Input(TransformSpec):
+    component_type: ClassVar[str]
+    default_source_path: ClassVar[Optional[str]]
+    def read(self) -> Any: ...
+    def __getattr__(self, name: str) -> Any: ...  # incomplete
+
+class Output(TransformSpec):
+    component_type: ClassVar[str]
+    default_destination_path: ClassVar[Optional[str]]
+    def __init__(
+        self,
+        destination: Optional[Any] = ...,
+        destination_path: Optional[Any] = ...,
+        encoding: Optional[str] = ...,
+        error_handler: str = ...,
+    ) -> None: ...
+    def write(self, data: str) -> Any: ...  # returns bytes or str
+    def encode(self, data: str) -> Any: ...  # returns bytes or str
+
+class FileInput(Input):
+    def __init__(
+        self,
+        source: Optional[Any] = ...,
+        source_path: Optional[Any] = ...,
+        encoding: Optional[str] = ...,
+        error_handler: str = ...,
+        autoclose: bool = ...,
+        mode: Literal["r", "rb"] = ...,
```

Comment:
This is the default but not the only possibility; it also allows `rU` in Python 2 at least.